Container path support

labkey-nicka opened the pull request:
#### Rationale
Add support for specifying the container path to `getUsersWithPermissions` and the Announcement module based components.

#### Related Pull Requests
* https://github.com/LabKey/labbook/pull/118
* https://github.com/LabKey/biologics/pull/1031
* https://github.com/LabKey/labkey-ui-components/pull/655
* https://github.com/LabKey/platform/pull/2714

#### Changes
* getUsersWithPermissions: support alternate container paths. Expose via `UserSelectInput` and `useUsersWithPermissions`.
* Announcements: support supplying containerPath.
* Update cache behind `getUsersWithPermissions` to cache a `Promise<List<User>>` rather than a `List<User>` so there isn't any interstitial resolution that could result in multiple requests for the same information.

labkey-nicka committed:
getUsersWithPermissions: support alternate container paths
- expose via UserSelectInput and useUsersWithPermissions

labkey-nicka committed:
Announcements: support supplying containerPath

labkey-nicka committed:
Update release notes

labkey-nicka committed:
Update ThreadEditor unit tests

labkey-alan commented on packages/components/src/internal/components/forms/actions.ts:
```diff
@@ -29,7 +29,7 @@ import {
 } from '../../..';
 
-import { getUsers, setUsers } from '../../global';
+import { getUsers, getUsersCacheKey, setUsers } from '../../global';
```
`getUsersCacheKey` is unused here.

labkey-nicka replied:
Fixed. Also, wanted to sure I mentioned ... "containerPath".

labkey-alan reviewed (approved):
Overall it looks good (did you type `containerPath` enough in this PR? 😂 ), I did find an unused import.

labkey-nicka committed:
Remove unused import

labkey-nicka committed:
QueryInfo: support containerPath in caching key
- deprecate invalidateQueryDetailsCacheKey in favor of invalidateQueryDetailsCache

labkey-nicka committed:
lint

labkey-nicka committed:
Update components.md

labkey-nicka committed:
v2.90.1-fb-subfolder-eln.0

labkey-nicka committed:
Prepare release

labkey-nicka committed:
v2.91.0
